pydantic/_internal/_generate_schema.py

```diff
@@ -821,10 +821,18 @@ def _generate_schema_from_property(self, obj: Any, source: Any) -> core_schema.C
         ):
             schema = existing_schema
         elif (validators := getattr(obj, '__get_validators__', None)) is not None:
-            warn(
-                '`__get_validators__` is deprecated and will be removed, use `__get_pydantic_core_schema__` instead.',
-                PydanticDeprecatedSince20,
-            )
+            from pydantic.v1 import BaseModel as BaseModelV1
+
+            if issubclass(obj, BaseModelV1):
+                warn(
+                    f'Nesting V1 models inside V2 models is not supported. Please upgrade `{obj.__name__}` to V2.',
+                    UserWarning,
+                )
```

Comment:
This will warn when doing the following:

```python
from pydantic import TypeAdapter
from pydantic.v1 import BaseModel as BaseModelV1

TypeAdapter(BaseModelV1)
```

So maybe there's a way to check for the current model stack before warning?

Reply:
Hmm, we don't support `TypeAdapter` with v1 models anyways... so maybe I should just make the warning more broad?

Reply:
I've modified the warning accordingly